src/events/neo4j.js

```diff
@@ -0,0 +1,386 @@
+    case 'Session':
+        connectionHolder = session._readConnectionHolder;
+        break;
+
+    case 'Transaction':
+        connectionHolder = session._connectionHolder;
+        break;
+
+    default:
+        return {
+            host, port,
+        };
+    }
+
+    if (session && connectionHolder && connectionHolder._connectionProvider) {
+        const connectionProvider = connectionHolder._connectionProvider;
+        port = connectionProvider._address.port();
+        host = connectionProvider._address.host();
+    }
+
+    return {
+        port, host,
+    };
+}
+
+
+/**
+ * Parse from DB session with Neo4j server, useful metadata
+ * @param {neo4j.Session|neo4j.Transaction} session A DB session (could be Session of Transaction)
+ * @returns {{metadata: Object, fullMetadata: Object}} JSONs with command's metadata and command's
+ *      full metadata which contains additional data about the event
+ */
+function getNeo4JSessionMetadata(session) {
+    const metadata = {};
+    const fullMetadata = {};
+
+    let connectionHolder;
+    let sessionMode;
+    let operationMode = '';
+
+    switch (session.constructor.name) {
+    // By default, Session opens two same connectionProviders - Read and Write, and not default one
+    case 'Session':
+        connectionHolder = session._readConnectionHolder;
+        sessionMode = 'READ|WRITE';
+        break;
+
+    case 'Transaction':
+        connectionHolder = session._connectionHolder;
+        sessionMode = connectionHolder._mode;
+
+        break;
+
+    default:
+        return;
+    }
+
+    switch (sessionMode) {
+    case 'READ':
+        operationMode = 'Read';
+        break;
+
+    case 'WRITE':
+        operationMode = 'Write';
+        break;
+
+    case 'READ|WRITE':
+        operationMode = 'Read|Write';
+        break;
+
+    default:
+        break;
+    }
+
+    if (connectionHolder) {
+        metadata.dbName = connectionHolder._database ? connectionHolder._database : 'default';
+        metadata.operation = `${session.constructor.name}:${operationMode}`;
+
+        fullMetadata.config = connectionHolder._connectionProvider._config;
+    }
+
+    // eslint-disable-next-line consistent-return
+    return { metadata, fullMetadata };
+}
+
+
+/**
+ * Create new patched methods for all relevant Observable methods and return
+ * @param {neo4j.session.Session | neo4j.transaction.Transaction} session A session
+ *      with Neo4j server
+ * @param {Object} observer An observer instance to patch
+ * @param {IArguments} sessionArguments The arguments of the observer
+ * @param {serverlessEvent.Event} dbApiEvent The event of the current run
+ * @param {number} startTime The Event start time
+ * @returns {Object} All the patched methods
+ */
+function getPatchedSubscribeMethods(
+    session,
+    observer,
+    sessionArguments,
+    dbApiEvent,
+    startTime
+) {
+    const records = [];
+
+    return {
+        // eslint-disable-next-line require-jsdoc
```

Comment:
what about the spread of ...observer
why did you remove it?

Reply:
By mistake, fixed, ty!